modules/repository-url/src/main/java/org/elasticsearch/common/blobstore/url/URLBlobStore.java

```diff
@@ -112,14 +113,14 @@ public void close() {
      * @return Base URL + path
      */
     private URL buildPath(BlobPath path) throws MalformedURLException {
-        String[] paths = path.toArray();
-        if (paths.length == 0) {
+        List<String> paths = path.parts();
+        if (paths.isEmpty()) {
             return path();
         }
-        URL blobPath = new URL(this.path, paths[0] + "/");
-        if (paths.length > 1) {
-            for (int i = 1; i < paths.length; i++) {
-                blobPath = new URL(blobPath, paths[i] + "/");
+        URL blobPath = new URL(this.path, paths.get(0) + "/");
+        if (paths.size() > 1) {
```

Comment:
This `if` isn't necessary, we check the condition in the loop anyway.

Reply:
++